File: Problems/Leetcode/3001-4000/3301-3400/3365.cpp
#include <bits/stdc++.h>

#include "debug.h"

using namespace std;

static constexpr int MOD = 1'000'000'007;

class Solution {
public:
    bool isPossibleToRearrange(string s, string t, int k) {
        map<string, int> mp;
        int len = s.size() / k;
        for (int i = 0; i < k; ++i) {
            mp[s.substr(i * len, len)]++;
            mp[t.substr(i * len, len)]--;
        }
        for (auto& [_, v] : mp) {
            if (v != 0) {
                return false;
            }
        }
        return true;
    }
};

int main() {
    Solution sol;

    return 0;
}

File: Problems/Leetcode/3001-4000/3301-3400/3365.py
class Solution:
    def isPossibleToRearrange(self, s: str, t: str, k: int) -> bool:
        n = len(s)
        k = n // k
        a = sorted(s[i : i + k] for i in range(0, n, k))
        b = sorted(t[i : i + k] for i in range(0, n, k))
        return a == b
